Source/URLRequest+Alamofire.swift

```diff
@@ -26,7 +26,7 @@ import Foundation
 
 public extension URLRequest {
     var method: HTTPMethod? {
-        guard let httpMethod = httpMethod else { return nil }
-        return HTTPMethod(rawValue: httpMethod)
+        get { return httpMethod.flatMap(HTTPMethod.init) }
+        set { httpMethod = newValue?.rawValue }
     }
```

Comment:
Not that we need to address this in this PR, but we should consider always returning a `.get` and not having this be optional. It defaults to `.get` even when not set.

Reply:
Huh. Not only does it default to `GET` but even after setting it to `nil` it still returns `GET`. What weird API. I can fix.

Reply:
Thinking about it more, since we only support a strict set of methods, it seems like this extension **should** return `nil` when the user has set a method string to a value we don't support.

Reply:
Oh great point! I really don't like the Apple API in this case.

Reply:
What does it even do if you set it to `INVALID` or something? Does it just fallback to `GET`?

Reply:
If it does just fall back to `GET`, then we probably can too.

Reply:
Nope, it returns the `String` entered, whatever it is.

Reply:
😩

Reply:
Optional it is 🤨